src/shared/capture/capture_video.cpp

```diff
@@ -0,0 +1,82 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+  \file    capture_video.cpp
+  \brief   C++ Implementation: CaptureVideo
+  \author  Felix Weinmann, (C) 2024
+*/
+//========================================================================
+
+#include "capture_video.h"
+
+#include <opencv2/imgproc.hpp>
+
+CaptureVideo::CaptureVideo(VarList* settings) : CaptureInterface(settings) {
+  settings->addChild(v_cap_file = new VarString("file", ""));
+  settings->addChild(v_cap_upscale = new VarBool("upscale", false));
+}
+
+
+RawImage CaptureVideo::getFrame() {
+  cv::Mat frame;
+  if(!capture.read(frame)) {
+    std::cout << "End of video stream reached" << std::endl;
+    return img;
+  }
+
+  int factor = v_cap_upscale->getBool() ? 2 : 1;
```

Comment:
Wouldn't it make more sense to configure the scale factor, instead of a flag?

Reply:
Other scalings should not be needed, this is just necessary due to the half resolution of the videos captured with the new VisionProcessor.

Reply:
Would be good to have some explanation of why this is necessary somewhere. I have not quite get why, yet :upside_down_face:

Reply:
Typical color cameras do an upscaling during the conversion from the Bayer matrix to the color image. So 4 sensor fields (RG - GB) are interpolated into 4 RGB pixels. To reduce necessary bandwidth and improve image processing time my new vision processor does an "downsampling" (only affects the green color channel) instead (RG - GB becomes 1 RGB pixel). To use these "half" resolution videos with the same geometry and blob resolution as the regular ssl-vision is used to this upscaling option is necessary.